TF-3712 Disallow inject as global

kevinzang opened the pull request:
##### *To:*
@udemy/team-f 

##### *What:*
Disallow inject as global.

BREAKING CHANGE: inject is no longer accepted as a global. Previously, we accepted it in angularjs specs, but it is problematic because ESLint doesn't complain when we use inject from mobx without importing it. fix angularjs specs via `import mocks from 'angularMocks'; mocks.inject`. See https://github.com/udemy/website-django/pull/31162.

##### *JIRA:*
https://udemyjira.atlassian.net/browse/TF-3712

##### *What did you test:*
I applied the change manually in website-django and fixed the lint errors in https://github.com/udemy/website-django/pull/31162.

##### *What dashboards will you be monitoring:*
None

Commit:
fix: disallow inject as global

BREAKING CHANGE: inject is no longer accepted as a global. previously, we accepted it in angularjs specs, but it is problematic because ESLint doesn't complain when we use inject from mobx without importing it. fix angularjs specs via `import mocks from 'angularMocks'; mocks.inject`.

cansin reviewed (approved):
LGTM. I wonder if `module` also Angular-specific?

kevinzang commented:
@cansin `module` isn't Angular-specific. I guess we wanted e.g. `module.exports = ...;` to be valid.

Commit:
Publish

 - babel-polyfill-udemy-website@8.0.6
 - babel-preset-udemy-website@9.0.6
 - eslint-config-udemy-website@10.0.0
